packages/2d/src/components/Latex.ts

```diff
@@ -0,0 +1,71 @@
+import {mathjax} from 'mathjax-full/js/mathjax';
+import {TeX} from 'mathjax-full/js/input/tex';
+import {SVG} from 'mathjax-full/js/output/svg';
+import {AllPackages} from 'mathjax-full/js/input/tex/AllPackages';
+import {liteAdaptor} from 'mathjax-full/js/adaptors/liteAdaptor';
+import {RegisterHTMLHandler} from 'mathjax-full/js/handlers/html';
+import {initial, signal} from '../decorators';
+import {Image, ImageProps} from './Image';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+import {OptionList} from 'mathjax-full/js/util/Options';
+import {useLogger} from '@motion-canvas/core/lib/utils';
+
+const adaptor = liteAdaptor();
+RegisterHTMLHandler(adaptor);
+
+const jaxDocument = mathjax.document('', {
+  InputJax: new TeX({packages: AllPackages}),
+  OutputJax: new SVG({fontCache: 'local'}),
+});
+
+export interface LatexProps extends ImageProps {
+  tex?: SignalValue<string>;
+  renderProps?: SignalValue<OptionList>;
+}
+
+export class Latex extends Image {
+  private static svgContentsPool: Record<string, string> = {};
+
+  private readonly imageElement = document.createElement('img');
+
+  @initial({})
+  @signal()
+  public declare readonly options: SimpleSignal<OptionList, this>;
+
+  @signal()
+  public declare readonly tex: SimpleSignal<string, this>;
+
+  public constructor(props: LatexProps) {
+    super(props);
+  }
+
+  protected override image(): HTMLImageElement {
+    // Render props may change the look of the TeX, so we need to cache both
+    // source and render props together.
+    const src = `${this.tex()}::${JSON.stringify(this.options())}`;
+    if (Latex.svgContentsPool[src]) {
+      this.imageElement.src = Latex.svgContentsPool[src];
+      return this.imageElement;
+    }
+
+    // Convert to TeX, look for any errors
+    const tex = this.tex();
+    const svg = adaptor.innerHTML(jaxDocument.convert(tex, this.options()));
+    if (svg.includes('data-mjx-error')) {
+      const errors = svg.match(/data-mjx-error="(.*?)"/);
+      if (errors && errors.length > 0) {
+        useLogger().error(`Invalid MathJax: ${errors[1]}`);
+      }
+    }
+    Latex.svgContentsPool[src] = svg;
```

Comment:
This will lack the header you add in line 63, so when the cache is used in line 47, the image will be invalid:
```ts
this.imageElement.src = Latex.svgContentsPool[src];
```
You can test that by rewinding the animation - it will work the first time, but any consecutive replays will crash. At least that's the case on my side.

I believe you should store the `text` variable in the cache:
```ts
    const text = `data:image/svg+xml;base64,${btoa(
      `<?xml version="1.0" encoding="UTF-8" standalone="no" ?>\n${svg}`,
    )}`;
    Latex.svgContentsPool[src] = text; // <- here
    const image = document.createElement('img');
```

Reply:
Good catch.